### ISSUE TITLE
Allow local Teleport users to authenticate with an SSO provider

### ISSUE THREAD

**Issue by pschisa** [gravitational/teleport#16014]
What would you like Teleport to do?
Allow for SSO providers to be linked to local Teleport users to provide authentication but not any further authorization or role mappings as is currently done with auth_connectors and SSO users

What problem does this solve?
Enable teleport to support the ability to provision/deprovision local teleport users externally(using the api) and use saml/oidc connectors strictly for authentication

If a workaround exists, please include it.
Not aware of any workaround

**Comment by zmb3** [comment]
Hmm, this seems very likely to cause confusion and complex configuration bugs. I wonder if there is a different way to address the concern.

**Comment by pschisa** [comment]
We did discuss the options of using role templates, SSO users, and API integration with the SSO as an alternative to still manage Teleport authorization from with APIs

**Comment by pschisa** [comment]
I think the difference is that this issue is focused on managing Teleport roles and users locally on the Teleport cluster via the API endpoint but being able to force those users to still authenticate via an SSO provider.

#9075 seems more focused on tracking the total number of SSO users in a system for an accurate estimate of user count on teleport when looking at users

**Comment by pschisa** [comment]
This may be an alternative approach to solve this use case for handling SSO integrations with Teleport: https://github.com/gravitational/teleport/pull/14977

**Comment by joshbalfour** [comment]
why the c*nt tag?

**Comment by zmb3** [comment]
@pschisa I think this is addressed by the Teleport SAML IdP. Can you confirm?

**Comment by djohns7** [comment]
@zmb3 I don't think this is covered through Teleport SAML or via login rules. The goal from the customer is to have local Teleport users stored in the platform with local policies, and then login to that locally stored user using SSO as the "password."
